src/search/FindInFiles.js

```diff
@@ -103,13 +110,31 @@ define(function (require, exports, module) {
             numMatchedLines  = matchedLines.length;
             totalMatchLength = match[0].length;
             lastLineLength   = matchedLines[matchedLines.length - 1].length;
+            endCh            = (numMatchedLines === 1 ? ch + totalMatchLength : lastLineLength);
+            highlightEndCh   = (numMatchedLines === 1 ? endCh : line.length);
+            highlightOffset  = 0;
             
-            // Don't store more than 200 chars per line
-            line = line.substr(0, Math.min(200, line.length));
+            if (highlightEndCh <= MAX_DISPLAY_LENGTH) {
+                // Don't store more than 200 chars per line
+                line = line.substr(0, Math.min(MAX_DISPLAY_LENGTH, line.length));
+            } else if (totalMatchLength > MAX_DISPLAY_LENGTH) {
+                // impossible to display the whole match
+                line = line.substr(ch, ch + MAX_DISPLAY_LENGTH);
+                highlightOffset = ch;
+            } else {
+                // Try to have both beginning and end of match displayed
+                padding = MAX_DISPLAY_LENGTH - totalMatchLength;
+                rightPadding = Math.floor(Math.min(padding / 2, line.length - highlightEndCh));
+                leftPadding = Math.ceil(padding - rightPadding);
+                highlightOffset = ch - leftPadding;
+                line = line.substring(highlightOffset, highlightEndCh + rightPadding);
+            }
             
             matches.push({
                 start:       {line: lineNum, ch: ch},
-                end:         {line: lineNum + numMatchedLines - 1, ch: (numMatchedLines === 1 ? ch + totalMatchLength : lastLineLength)},
+                end:         {line: lineNum + numMatchedLines - 1, ch: endCh},
```

Comment:
@MarcelGerber I think this might run into some bugs with matches that span multiple lines.  The code here still assumes the start ch is on the first line of the match and the end ch is on the last, which might be incorrect depending on which clipping case you fall into.  This was already somewhat buggy before your change, so it may not be an urgent issue... but just fyi.  In case you feel like doing any more work on this code :-)

Also, might not hurt to add some clipping tests to FindInFiles-test at some point.